pkg/api/server.go

```diff
@@ -53,12 +54,13 @@ func handleStatus(w http.ResponseWriter, r *http.Request) {
 
 // CreateAppParams holds all the necessary fields to create a new ffaas application.
 type CreateAppParams struct {
-	Name string `json:"name"`
+	Name        string            `json:"name"`
+	Environment map[string]string `json:"environment"`
 }
 
 func (p CreateAppParams) validate() error {
 	if len(p.Name) < 3 || len(p.Name) > 20 {
```

Comment:
If the error message is changed to 'less than 40 characters' then so should be the conditional test. 
It should be `if len(p.Name) < 3 || len(p.Name) > 40`